samples/oauth_rails_vethealth/app/models/authentication.rb

```diff
@@ -0,0 +1,53 @@
+class Authentication < ApplicationRecord
+  serialize :id_token
+  belongs_to :oauth_callback, optional: true
+
+  attr_reader :session_errors
+
+  def self.generate_nonce(base)
+    Digest::SHA256.hexdigest(base + ENV['va_developer_client_secret'])
+  end
+
+  def self.attributes_from_oauth(response)
+    attributes_array = response.map do |key,value|
```

Comment:
If you are only replacing the 'expires_at' value, then I think mapping all of the key value pairs is excessive. Perhaps do this instead:
`attributes_array = response` (I suppose you could dup here if that's an issue)
`attributes_array[:expires_at] = Time.zone.at(response[:expires_at])`